crates/paths/src/lib.rs

```diff
@@ -0,0 +1,321 @@
+                data_dir: server::ServerDataDir(base_dirs.get_data_file("data")),
+            })
+        }
+    }
+
+    pub fn from_root_dir(dir: &RootDir) -> Self {
+        Self {
+            cli_config_dir: dir.cli_config_dir(),
+            cli_bin_file: dir.cli_bin_file(),
+            cli_bin_dir: dir.cli_bin_dir(),
+            data_dir: dir.data_dir(),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::ffi::{OsStr, OsString};
+    use std::path::Path;
+
+    use super::*;
+
+    mod vars {
+        use super::*;
+        struct ResetVar<'a>(&'a str, Option<OsString>);
+        impl Drop for ResetVar<'_> {
+            fn drop(&mut self) {
+                maybe_set_var(self.0, self.1.as_deref())
+            }
+        }
+        fn maybe_set_var(var: &str, val: Option<impl AsRef<OsStr>>) {
+            if let Some(val) = val {
+                std::env::set_var(var, val);
+            } else {
+                std::env::remove_var(var);
+            }
+        }
+        pub(super) fn with_vars<const N: usize, R>(vars: [(&str, Option<&str>); N], f: impl FnOnce() -> R) -> R {
+            let _guard = vars.map(|(var, val)| {
+                let prev_val = std::env::var_os(var);
+                maybe_set_var(var, val);
+                ResetVar(var, prev_val)
+            });
+            f()
+        }
+    }
+
+    #[cfg(not(windows))]
+    #[test]
+    fn xdg() {
+        let p = Path::new;
+        let paths = vars::with_vars(
+            [
+                ("XDG_CONFIG_HOME", Some("/__config_home")),
+                ("XDG_DATA_HOME", Some("/__data_home")),
+                ("XDG_BIN_HOME", Some("/__bin_home")),
+            ],
+            SpacetimePaths::platform_defaults,
+        )
+        .unwrap();
+        assert_eq!(paths.cli_config_dir.0, p("/__config_home/spacetime"));
+        assert_eq!(paths.cli_bin_file.0, p("/__bin_home/spacetime"));
+        assert_eq!(paths.cli_bin_dir.0, p("/__data_home/spacetime/bin"));
+        assert_eq!(paths.data_dir.0, p("/__data_home/spacetime/data"));
+    }
+
+    #[cfg(windows)]
+    #[test]
+    fn windows() {
+        let paths = SpacetimePaths::platform_defaults().unwrap();
+        let appdata_local = dirs::data_local_dir().unwrap();
+        assert_eq!(paths.cli_config_dir.0, appdata_local.join("config"));
+        assert_eq!(paths.cli_bin_file.0, appdata_local.join("spacetime.exe"));
+        assert_eq!(paths.cli_bin_dir.0, appdata_local.join("bin"));
+        assert_eq!(paths.data_dir.0, appdata_local.join("data"));
+    }
+
+    #[test]
+    fn custom() {
+        let root = Path::new("/custom/path");
+        let paths = SpacetimePaths::from_root_dir(&RootDir(root.to_owned()));
+        assert_eq!(paths.cli_config_dir.0, root.join("config"));
+        assert_eq!(
+            paths.cli_bin_file.0,
+            root.join("spacetime.exe").with_extension(EXE_EXTENSION)
+        );
+        assert_eq!(paths.cli_bin_dir.0, root.join("bin"));
```

Comment:
Why assume `custom` is for Windows? 

BTW I have [this solution](https://github.com/clockworklabs/SpacetimeDB/blob/79d1654add17a3a99e80dfd5cc42a90c11ba3507/crates/platforms/src/directories.rs#L129) to make it work everywhere.

Reply:
Because that's what the proposal is - it doesn't say it outright, but it's directly equivalent to a RootDir based at `%LocalAppData%/SpacetimeDB`. So, why duplicate the implementation if it's no different?